### ISSUE TITLE
Slow?

### ISSUE THREAD

**Issue by techpulse** [fail2ban/fail2ban#1183]
I was under a quiet large botnet attack, somehow. The IP, managed to attack me by reaching the limit 1600 times (even though I made it ban after three tries) http://i.imgur.com/AdgZMNH.png

how is this possible, how can i make it ban faster or scan the error.log faster?

I used https://rtcamp.com/tutorials/nginx/fail2ban/ (their code is fine, however it's fail2ban that.. is slow..?? or my CPU??)

**Comment by sebres** [comment]
In the fail2ban self you can control a ban only via `maxretry` - maximum failures occurred in log during `findtime` seconds. If you set `maxretry = 1` this ip goes ban already after 1st matched line in nginx/error.log file.
But the time when this line will be written controls the nginx self.
Either you should make expected changes in your configuration of nginx `nginx.conf` to write this entries faster (using `limit_conn` or `limit_req`, if you speak Russian - see here http://habrahabr.ru/search/?q=nginx+fail2ban+limit_req&target_type=posts&order_by=relevance).
Or if you mean, the nginx writes it fast enough, but the fail2ban reads a log to slow, you can try my branch #1116 that makes fail2ban faster (and hereafter post the result here:).

**Comment by Octolus** [comment]
Any updates on this @sebres 

I notice it takes 3 seconds to ban when parsing nginx log, is there a way to lower that time? I have a really powerful server.

**Comment by sebres** [comment]
> Any updates on this @sebres

What do you mean exactly? Many of performance issues are fixed in 0.10/0.11.
So may be you can update your fail2ban version.
Which fail2ban version do you use at the moment?

> I notice it takes 3 seconds to ban when parsing nginx log...

Why you ask us and not cloudflare?
Just kidding, but... 
If I were you, I'd try to use some more performant action banning directly on your host (and if you need cloudflare too, maybe as second action)...
So you can configure IP-related ban-action (like `iptables` etc) banning via firewall of your choice or nginx-related actions like [nginx-block-map.conf](https://github.com/fail2ban/fail2ban/blob/0.10/config/action.d/nginx-block-map.conf) banning directly over nginx.
```ini
[luaban]
# firstly ban locally then notify cloudflare:
action = iptables-multiport[...parameters...]
         cloudflare[...parameters...]
```
But I'd like to know, what exactly 3 seconds are?
Which time do you see in the ddos-log which causes ban and in fail2ban.log for this ticket (`[luaban] Ban ...`). Are you really see 3 seconds between both?
Also note some connects may be keep-alive, if you don't reject connection explicitly (in your lua-ddos scripts, nginx-location or banning action e.g. via firewall) the intruder may generated several requests before it gets banned in cloudflare.
